aptos-move/block-executor/src/executor.rs

```diff
@@ -1445,12 +1429,11 @@ where
         signature_verified_block: &[T],
         base_view: &S,
     ) -> Result<BlockOutput<E::Output>, E::Error> {
-        let dynamic_change_set_optimizations_enabled = signature_verified_block.len() != 1
-            || E::is_transaction_dynamic_change_set_capable(&signature_verified_block[0]);
+        // TODO[agg_v2](cleanup) - this now just sets based on capability, but
+        // we can instead use the feature flag to set whether it is actually enabled.
+        let dynamic_change_set_optimizations_enabled = true;
```

Comment:
Why? This means we always do separate group processing, etc. in sequential?

Reply:
this sets the capability (is_delayed_field_optimization_capable is the view interface). if we get onchain configs into BlockSTM, we could set the enabled here instead. 

I can rename the variable to capable, until TODO is addressed

Reply:
I see, thanks, I think we better land this with TODO first

Reply:
So I also got thoroughly confused but unlike George, couldn't unconfuse myself.
1. Parallel view always returning optimization capable now, but sequential is based on this flag.
2. Sequential also gates resource group and delayed field processing on this flag, which seems to be merely an optimization: since it's set in the view to not be capable we wouldn't expect to receive any granular groups or delayed fields. If this is the case, can we also assert the corresponding containers are empty? And moreover, the optimization isn't really saving much, mainly iterating over empty vectors?
3. Currently none of them have effect since output doesn't have delayed fields and resource groups: in fact parallel always does the "empty" processing that is 
`if dynamic_change_set_optimization_enabled {`
-gated in sequential execution? (maybe an argument to remove that if?)

However, I can't reconcile the above with 
(a) how is this related to direct writeset (do we need to configure it with sequential execution and is it important to have view return is_delayed_field_optimization_capable = false? if not, then why does sequential need to be different?).
(b) the comment says we can set this based on the feature flag, but shouldn't the feature flag control the output preparation instead (for replaying, etc), o.w. why would the feature flag affect a variable here that only affects sequential execution?

Reply:
This PR cleans up some vestiges of trying to do double-fallback, but not all because we can make it useful later. But I can completely remove it for now as it is confusing you all.

What it was meant to be used later: imagine if BlockExecutor could get the information if the feature flag is set or not. then it could decide whether to call incorporate_materialized_txn_output or set_txn_output_for_non_dynamic_change_set (and second one asserts no  unexpected writes are present in the output). 

Issue is, it is not easy for BlockExecutor to get this feature flag information through constructor.